### DISCUSSION TITLE
Disable encoding and show only original file.

### DISCUSSION THREAD

**Discussion by Zibbp** [mediacms-io/mediacms#285] [Q&A]
Hello,

I'm using the docker setup - is it possible to disable encoding and display the original video when viewing a video?

Thanks.

**Comment by swiftugandan** [comment]
I think if you removed/disabled all the encoding profiles, encoding would be disabled

**Reply by Zibbp** [reply]
I thought that as well and I've tried disabling all encoding profiles or disabling all except the GIF preview and when uploading media it is forever marked as "Media encoding in progress".

**Comment by mgogoulos** [comment]
Hi @Zibbp , this is not possible,  besides serving different resolutions (240, 1080 etc) video transcoding through H264 (that is enabled by default) ensures that the transcoded version of a video will play on a number of clients (older browsers, mobile) that the original video won't play. 

You may choose to disable all encoding profiles as @swiftugandan suggests, and then let the API get the original file instead of the transcoded versions, I can send you more hints if this interests you.

Out of curiosity, what would be the motivation to play original files and not transcoded versions?

**Reply by swiftugandan** [reply]
@mgogoulos one of the use cases of this sort of thing is if you don't want to do any transcoding on the device (which may be of low specs) but still want to serve the original videos.

**Reply by Zibbp** [reply]
@mgogoulos I've tried disabling all encoding profiles but when uploading a new video it is forever marked as "Media encoding in progress". Is there a simple change I can make to allow the playback of the original video?

My motivation for only having the original video is to use mediacms as a backend/frontend for my large Youtube archive. I don't need to support multiple qualities and doing so would only be wasting space.

**Reply by kajatak** [reply]
Would it be possible to get some more details regarding what needs to be changed in the API? I would also like to use MediaCMS for a personal YouTube archive as no other alternative fits my requirements, but I have no need to re-encode all the files that I'd like to add as it would only be used by myself on a desktop computer.

**Reply by mgogoulos** [reply]
Checkout branch `feat-original-exp` https://github.com/mediacms-io/mediacms/tree/feat-original-exp

edit `cms/settings.py` and set 

```
DO_NOT_TRANSCODE_VIDEO = True
```

What this will do: when uploading a video file, it won't create any encoding, and the player will show the original video. It's very ugly but you try it and it provides value, we can consider how it can enter the main branch

**Reply by kajatak** [reply]
Thank you, this seems to be working perfectly.

**Reply by falense** [reply]
I really like this feature too. Please add it to the main release. For now I have applied the changed to latest manually.

**Reply by foeyonghai** [reply]
@mgogoulos I just tested this by using the main branch and change the settings from your commit by hand. That works like a charm. Nevertheless we are 2 years further already, is their a certain reason that this is still not in the main branch? Your solution is perfect!

**Reply by mgogoulos** [reply]
Thanks for the reminder, and great to know it works without issues, I am going to merge, close or comment on all open PRs one of these days and this one will find it's way!

**Reply by foeyonghai** [reply]
@mgogoulos I see that is it merged now, super! This saves me a great amount of processing time and storage, even though it is on my local servers and does not run in the cloud this is a great improvement!

**Comment by devilmind81** [comment]
I also need to have only the original file and this option should be very good for me.
Could you support me telling  how can I integrate the changes your released in the feauture branch in the main one?
Which files I must update?

Thanks in advance

**Comment by streamly** [comment]
Is the DO_NOT_TRANSCODE_VIDEO = True in the latest install version? I just installed it and added this option, restarted and videos are still encoding. thx

**Reply by mgogoulos** [reply]
Unfortunately, it's not!

**Comment by 3rd-st-ninja** [comment]
Can this feature be merged in?  Is there any reason it isn't or shouldn't be?

**Comment by marcofloriano** [comment]
Wating for this feature too. It gives MediaCMS a new form of life, so i can manually transcode and upload my videos elsewhere (vimeo on my case) and save tons of space, since i will provide the original file only via API.

**Reply by mgogoulos** [reply]
It's like a hack, but since it's requested I've opened a work in progress PR on https://github.com/mediacms-io/mediacms/pull/829 and will try to test it locally and see if it needs something or it can be merged...

**Comment by mgogoulos** [comment]
PR is now merged, checkout the note on https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md
